executor/show.go

```diff
@@ -2039,9 +2047,16 @@ func (e *ShowExec) fetchShowTableRegions(ctx context.Context) error {
 		if indexInfo == nil {
 			return plannercore.ErrKeyDoesNotExist.GenWithStackByArgs(e.IndexName, tb.Meta().Name)
 		}
-		regions, err = getTableIndexRegions(indexInfo, physicalIDs, tikvStore, splitStore)
+		if indexInfo.Global {
+			regions, err = getTableIndexRegions(indexInfo, []int64{tb.Meta().ID}, tikvStore, splitStore)
+		} else {
+			regions, err = getTableIndexRegions(indexInfo, physicalIDs, tikvStore, splitStore)
+		}
 	} else {
 		// show table * region
+		if hasGlobalIndex {
+			physicalIDs = append([]int64{tb.Meta().ID}, physicalIDs...)
```

Comment:
I wonder if this is more efficient and easier to read?
```suggestion
			physicalIDs = append(physicalIDs, tb.Meta().ID)
```

Reply:
@mjonss Thanks for your review~
This slice element sequence affects the display of regions. 
I think both styles are OK, maybe setting logical table in front of the partitioned table is nicer for readers. If you think the later efficient one is better, I will propose a new PR.

Reply:
Ah, that is an OK reason, to have the table level regions first.